### PR TITLE
Sampling apply to all edges starting from a vertex.

### PR DESCRIPTION
**Pull request by CPWstatic**

### CODE REVIEW

File: src/storage/query/QueryBaseProcessor.inl [hunk 1]
```diff
@@ -491,15 +491,6 @@ kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectEdgeProps(
...
     Getters getters;
-    std::unique_ptr<nebula::algorithm::ReservoirSampling<
-        std::pair<std::unique_ptr<RowReader>, std::string>>> sampler;
-    if (FLAGS_enable_reservoir_sampling) {
-        sampler = std::make_unique<
-            nebula::algorithm::ReservoirSampling<
-                std::pair<std::unique_ptr<RowReader>, std::string>
-            >
-        >(FLAGS_max_edge_returned_per_vertex);
-    }
```
Comment (panda-sheep) [inline src/storage/query/QueryBaseProcessor.inl]: 498  line  need handle ?

File: src/storage/query/QueryBoundProcessor.cpp [hunk 2]
```diff
@@ -35,17 +35,19 @@ kvstore::ResultCode QueryBoundProcessor::processEdgeImpl(const PartitionID partI
     edges.reserve(FLAGS_reserved_edges_one_vertex);
     auto ret = collectEdgeProps(
         partId, vId, edgeType, props, &fcontext,
-        [&, this](RowReader* reader, folly::StringPiece k, const std::vector<PropContext>& p) {
+        [&, this](std::unique_ptr<RowReader> reader,
+                  folly::StringPiece k,
+                  const std::vector<PropContext>& p) {
             cpp2::IdAndProp edge;
```
Comment (dangleptr) [inline src/storage/query/QueryBoundProcessor.cpp]: Could we unify the code here?
```
auto ret = sample(xxx)
if (ret) {
     xxxx
}
```

**Comment by jude-zhu** [comment]
close  #1982

File: src/storage/query/QueryBaseProcessor.inl (hunk 1 above)
Comment (CPWstatic) [reply src/storage/query/QueryBaseProcessor.inl:494]: We need it, or we would not the whole samples.

File: src/storage/query/QueryBoundProcessor.cpp (hunk 2 above)
Comment (CPWstatic) [reply src/storage/query/QueryBoundProcessor.cpp:41]: As we talked about this offline.

**Pull request merged**